content/trust-management/trust-authority.md

```diff
@@ -60,23 +56,163 @@ its list of public signature verification keys by removing the compromised key
+
+
+* `blacklistedEntities` is a map associating a Matrix user ID to a JSON object
+  using the following structure:
+
+|      Parameter      |        Type       | Req. |                                                 Description                                                 |
+| ------------------- | ----------------- | :--: | ----------------------------------------------------------------------------------------------------------- |
+| `reason_code`       | `string`          |  x   | One of the reason codes defined [above](#blacklist-reason-codes).                                           |
+| `after`             | `string`          |  x   | ID of the latest trustworthy event sent by the entity. Empty string if none (e.g. with `B_MISINFORMATION`). |
+| `additional_info`   | `localisedString` |      | More information on the blacklist.                                                                          |
+
+#### Example
+
+```json
+{
+    "name": {
+        "en": "Some NGO",
+        "fr": "Une ONG"
+    },
+    "sig_algo": "ed25519",
+    "sig_keys": [
+        "IlRMeOPX2e0MurIyfWEucYBRVOEEUMrOHqn/8mLqMjA"
+    ],
+    "website": "https://www.somengo.org",
+    "description": {
+        "en": "We do activism for freedom of the press.",
+        "fr": "Nous sommes des activistes en faveur de la liberté de la presse."
+    },
+    "logo": "mxc://weu.informo.network/AtEuTuVSeIlZQgjEzjGyMDtG",
+    "trusted": {
+        "@acmenews:example.com": {
+            "signature": "0a1df56f1c3ab5b1",
+            "type": "source"
+        },
+        "@someothersource:example2.com": {
+            "signature": "daiRanaiy1be7pe",
+            "type": "source"
+        }
+    },
+    "blacklist": {
+        "@sadsource:example.com": {
+            "reason_code": "B_COMPROMISED",
+            "since": "!someEvent:example.com"
+        },
+        "@badsource:example.com": {
+            "reason_code": "B_MISINFORMATION",
+            "since": "",
+            "additional_info": {
+                "en": "This source actually belongs to the Tomainian government which uses it for propaganda.",
+                "fr": "Cette source appartient au gouvernement Tomanien qui l'utilise à des fins de propagande."
+            }
+        }
+    }
+}
+```
+
+## Blacklisting
+
+A trust authority **can** list sources and trust authorities that it explicitly
+blacklists for being compromised or ethical reasons. The trust authority
+**must** specify a reason for blacklisting a source or TA, which consists in a
+defined reason code that client implementations **can** use in order to warn
+users about a specific source. The trust authority **can** also provide
+additional information to explain the addition to the blacklist, which, if
+provided, **must** be contained in a custom localised string. Client
```

Comment:
I'd rather say: `The trust authority **can** also provide additional information to explain the addition to the blacklist, as a localisedString map.`, or something similar, since IMHO the additional information "is" the localisedString map, rather than "stored inside" a localisedString.

Also the "must" is not very useful since it is already defined as a localisedString the data structure.

Reply:
I don't like the idea of naming "internal" types such as `localisedString` outside of data structure definitions.

> Also the "must" is not very useful since it is already defined as a localisedString the data structure.

True, but I find that specifying its form makes the definition more complete.